rasa/nlu/classifiers/embedding_intent_classifier.py

```diff
@@ -963,84 +942,66 @@ def _prepare_layers(self) -> None:
                 self.config[UNIDIRECTIONAL_ENCODER],
                 name="text_encoder",
             )
-        else:
-            self._transformer = lambda x, mask, training: x
-
-        self._embed = {}
-
-        self.train_metrics = {"t_loss": tf.keras.metrics.Mean(name="t_loss")}
-        self.eval_metrics = {"val_t_loss": tf.keras.metrics.Mean(name="val_t_loss")}
+            if self.config[NUM_TRANSFORMER_LAYERS] > 0
+            else lambda x, mask, training: x
+        )
 
-        if self.config[MASKED_LM]:
-            self._input_mask = tf_layers.InputMask()
-            self._embed["text_mask"] = tf_layers.Embed(
-                self.config[EMBED_DIM],
-                self.config[C2],
-                "text_mask",
-                self.config[SIMILARITY_TYPE],
-            )
-            self._embed["text_token"] = tf_layers.Embed(
-                self.config[EMBED_DIM],
-                self.config[C2],
-                "text_token",
-                self.config[SIMILARITY_TYPE],
-            )
-            self._loss_mask = tf_layers.DotProductLoss(
-                self.config[NUM_NEG],
-                self.config[LOSS_TYPE],
-                self.config[MU_POS],
-                self.config[MU_NEG],
-                self.config[USE_MAX_SIM_NEG],
-                self.config[C_EMB],
-                self.config[SCALE_LOSS],
-            )
-            self.train_metrics["m_loss"] = tf.keras.metrics.Mean(name="m_loss")
-            self.train_metrics["m_acc"] = tf.keras.metrics.Mean(name="m_acc")
-            self.eval_metrics["val_m_loss"] = tf.keras.metrics.Mean(name="val_m_loss")
-            self.eval_metrics["val_m_acc"] = tf.keras.metrics.Mean(name="val_m_acc")
-        else:
-            self._input_mask = None
-            self._loss_mask = None
+    def _prepare_mask_lm_layers(self) -> None:
+        self._tf_layers["input_mask"] = tf_layers.InputMask()
+        self._tf_layers["embed.lm_mask"] = tf_layers.Embed(
+            self.config[EMBED_DIM],
+            self.config[C2],
+            "text_mask",
```

Comment:
```suggestion
            "lm_mask",
```
to keep the names consistent